src/sentry/api/serializers/models/rule.py

```diff
@@ -289,3 +302,192 @@ def serialize(self, obj, attrs, user, **kwargs) -> RuleSerializerResponse:
             d["disableDate"] = attrs["disable_date"]
 
         return d
+
+
+class WorkflowEngineRuleSerializer(Serializer):
+    def __init__(
+        self,
+        expand: list[str] | None = None,
+        prepare_component_fields: bool = False,
+        project_slug: str | None = None,
+    ):
+        super().__init__()
+        self.expand = expand or []
+        self.prepare_component_fields = prepare_component_fields
+        self.project_slug = project_slug
+
+    def get_attrs(self, item_list: Sequence[Workflow], user, **kwargs):
```

Comment:
can we decompose this method a bit? it'd be nice to break it apart by each attribute. if you order them in the right ways you can probably keep it pretty clean.

Reply:
i've decomposed it 💀